### PR TITLE
Define package body symbol

### PR DESCRIPTION
**Pull request by xbauch**
This is a refactor of #230 with updated golden results. Copy of its description:

In Do_Compilation_Unit the return value Unit_Symbol was not set for a package body

To determine a symbol value for a package body a symbol is determined from
its specification and registered in the symbol table.  This required checking for
the package being a child.

Note: The same check may be required for child subprograms!

### CODE REVIEW

File: gnat2goto/driver/tree_walk.adb [hunk 1]
```diff
@@ -954,29 +956,26 @@ package body Tree_Walk is
...
+            --  The specification of the subprogram body has already
+            --  been inserted into the symbol table by the call to
+            --  Do_Withed_Unit_Specs.
+            pragma Assert (Global_Symbol_Table.Contains (Unit_Name));
+            Unit_Symbol := Global_Symbol_Table (Unit_Name);
 
-               --  Now compile the body of the subprogram
-               Unit_Symbol.Value := Do_Subprogram_Or_Block (U);
+            --  Now compile the body of the subprogram
+            Unit_Symbol.Value := Do_Subprogram_Or_Block (U);
```
Comment (hannes-steffenhagen-diffblue) [inline gnat2goto/driver/tree_walk.adb]: Is this supposed to be the elaboration code for the package?

Comment (tjj2017) [reply gnat2goto/driver/tree_walk.adb]: No.  It just gives a valid symbol.  In the code for Do_Compilation_Unit this symbol is inserted into the symbol table.  I did not try to rewrite the existing code of Do_Compilation_Unit, just provide a valid symbol to insert.

The elaboration code needs further thought.  We need to provide initialisation code which can be called prior to calling the designated main program.

**Comment by hannes-steffenhagen-diffblue** [comment]
I still don't quite understand what we need a package symbol in the symbol table for

**Comment by chrisr-diffblue** [comment]
@xbauch @tjj2017 - Does this PR replace/supeceed/subsume #230, or is it just dependent on #230? If it replaces #230 could you please close the #230 PR - or alternatively, if *this* PR has now been merged into #230, could this one be closed :-) I'd just like to be clear one which of these two PRs needs to be merged :-)

**Comment by tjj2017** [comment]
This PR supersedes PR #230. I have rebased this branch with the latest diff/blue master.

**Comment by xbauch** [comment]
@chrisr-diffblue and @tjj2017 Well I rebased (and updated the golden results) but @hannes-steffenhagen-diffblue's question is still open, i.e. why do we need the package symbol in the symbol table?

**Comment by chrisr-diffblue** [comment]
In the interests of reducing the backlog of PRs we have, I think we should merge this and then discuss/resolve the question of why/whether we need the package symbol in the symbol table separately.

**Pull request merged**